### DISCUSSION TITLE
MPU-6050 (GY-521)

### DISCUSSION THREAD

**Discussion by geekpaolo** [arendst/Tasmota#15400] [Q&A] [answered]
I need to use gyro. 
On Arduino/EP8266 if I use library like GY-521 (https://www.arduino.cc/reference/en/libraries/gy521) and works fine both with ESP as Arduino. 

If I use sensor (Tasmota/tasmota/xsns_32_mpu6050.ino),  measures are not stable ("dance") :-(.

I see that Tasmota has a library (/lib/lib_i2c/I2Cdevlib-MPU6050). Which is the difference with the sensor above?
Which I have to use to have more stable measure?

Thanks in advance

**Comment by sfromis** [comment, marked as answer]
It is not either-or, as Tasmota only has one option; `xsns_32_mpu6050.ino` is using `I2Cdevlib-MPU6050`
